### PR TITLE
Fixed code generation when using arrays in Value Type types.

### PR DESCRIPTION
**Pull request by genuinelucifer**
Fixed all the changed in one commit and a mergeable PR...

### CODE REVIEW

File: src/Generator/Generators/CSharp/CSharpTextTemplate.cs [hunk 1]
```diff
@@ -856,12 +856,30 @@ private void GeneratePropertySetter<T>(QualifiedType returnType, T decl, Class @
...
                 var marshal = new CSharpMarshalManagedToNativePrinter(ctx);
-                ctx.ReturnVarName = string.Format("{0}{1}{2}",
+                
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    string arrPtrIden = Helpers.SafeIdentifier("__arrPtr");
```
Comment (ddobrev) [inline src/Generator/Generators/CSharp/CSharpTextTemplate.cs]: This works too but what I meant was that you could just do:

Helpers.SafeIdentifier("arrPtr")

The point is that SafeIdentifier prepends "__" for you, that's the point. No problem if you leave it this way, I just wanted you to know what the convenience of SafeIdentifier.

File: src/Generator/Generators/CLI/CLIMarshal.cs [hunk 2]
```diff
@@ -40,13 +40,15 @@ public override bool VisitTagType(TagType tag, TypeQualifiers quals)
             return decl.Visit(this);
         }
 
+        private static int arrayNum = 0;
```
Comment (ddobrev) [inline src/Generator/Generators/CLI/CLIMarshal.cs]: I believe I've already sent you a remark about this. A static value keeps increasing for the lifetime of the type - that is, for the whole run of the application. This means that for each new array ANYWHERE you'll increase this index.
We already have a mechanism to deal with this. Context has properties called ParameterIndex and MarshalVarPrefix. See how they are used and employ them in your code.

File: src/Generator/Generators/CSharp/CSharpTextTemplate.cs (hunk 1 above)
Comment (genuinelucifer) [inline src/Generator/Generators/CSharp/CSharpTextTemplate.cs]: Oh sorry I changed that without actually looking at Helpers.SafeIdentifier

File: src/Generator/Generators/CLI/CLIMarshal.cs (hunk 2 above)
Comment (genuinelucifer) [inline src/Generator/Generators/CLI/CLIMarshal.cs]: Thanks. I implemented something similar in CSharpTextTemplate but could not do same here. I will look into ParameterIndex and try to implement.

**Comment by genuinelucifer** [comment]
Due to latest Merges this PR has now conflict merges and is incompilable on my machine too since i pulled the changes. So I am closing it and opening a new one with all the required changes.

**Pull request closed without merging**